SofaKernel/framework/sofa/simulation/MechanicalMatrixVisitor.h

```diff
@@ -223,17 +223,6 @@ class SOFA_SIMULATION_CORE_API MechanicalAddMBK_ToMatrixVisitor : public Mechani
         return RESULT_CONTINUE;
     }
 
-    //Masses are now added in the addMBKToMatrix call for all ForceFields
-
-    virtual Result fwdProjectiveConstraintSet(simulation::Node* /*node*/, core::behavior::BaseProjectiveConstraintSet* c)
```

Comment:
Why this has been removed ?
Is it replaced by MechanicalApplyProjectiveConstraint_ToMatrixVisitor ?

Reply:
Yes, it's related to
> MechanicalMatrixVisitor and MechanicalOperations: fix if using a Linear Solver, projective constraints were wrongly applied when a force field is in a child node

When you have a MechanicalObject and a projective constraint in a (parent) node and a ForceField in a child node, which you have to do if for instance you want to apply it to the DOFs but with a different topology for the elements (a subset, or converting hexas to tetras for example), then the matrix was projected before the forcefield would add its contributions.
In order to fix it in all cases (allowing for both forcefield and projective constraints in child nodes) the solution is now to do it with two visitors.

Reply:
Yes, it's related to
> MechanicalMatrixVisitor and MechanicalOperations: fix if using a Linear Solver, projective constraints were wrongly applied when a force field is in a child node

When you have a MechanicalObject and a projective constraint in a (parent) node and a ForceField in a child node, which you have to do if for instance you want to apply it to the DOFs but with a different topology for the elements (a subset, or converting hexas to tetras for example), then the matrix was projected before the forcefield would add its contributions.
In order to fix it in all cases (allowing for both forcefield and projective constraints in child nodes) the solution is now to do it with two visitors.